adaptation.lua

```diff
@@ -0,0 +1,110 @@
+-- Original opening comment before lua adaptation
+-- View or set cavern adaptation levels
+-- based on removebadthoughts.rb
+-- Rewritten by TBSTeun using OpenAI GPT from adaptation.rb
+
+local function print_color(color, s)
+    dfhack.color(color)
+    dfhack.print(s)
+    dfhack.color(COLOR_RESET)
+end
+
+local function usage(s)
+    if s then
+        print(s)
+    end
+    print(dfhack.script_help())
+end
+
+local function set_adaptation_value(unit, v)
+    local num_set = 0
+
+    if not dfhack.units.isCitizen(unit) or unit.flags2.killed then
+        return num_set
+    end
+    local trait_found = false
+    for _, t in ipairs(unit.status.misc_traits) do
+        if t.id == df.misc_trait_type.CaveAdapt then
+            if mode == "show" then
+                    print_color(COLOR_RESET, "Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") has an adaptation of ")
+                    if t.value <= 399999 then
+                        print_color(COLOR_GREEN, t.value .. "\n")
+                    elseif t.value <= 599999 then
+                        print_color(COLOR_YELLOW, t.value .. "\n")
+                    else
+                        print_color(COLOR_RED, t.value .. "\n")
+                    end
+            elseif mode == "set" then
+                print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)).. ") changed from " .. t.value .. " to " .. v)
+                t.value = v
+                num_set = num_set + 1
+            end
+            trait_found = true
+            break
+        end
```

Comment:
```suggestion
        if t.id ~= df.misc_trait_type.CaveAdapt then goto continue end
        if mode == "show" then
            print_color(COLOR_RESET, ('Unit %d (%s) has an adaptation of '):format(unit.id, dfhack.TranslateName(dfhack.units.getVisibleName(unit)))
            if t.value <= 399999 then
                print_color(COLOR_GREEN, t.value .. "\n")
            elseif t.value <= 599999 then
                print_color(COLOR_YELLOW, t.value .. "\n")
            else
                print_color(COLOR_RED, t.value .. "\n")
            end
            return 0
        end
        print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)).. ") changed from " .. t.value .. " to " .. v)
        t.value = v
        return 1
        ::continue::
    end
```

Reply:
Lua doesn't have a formal `continue` statement. Instead there is an idiom using `goto` and a "continue" label.

Reply:
(btw I left the formatting of the second print statement as an exercise for the reader : )

Reply:
Assuming you mean string concatenation with your print statement comment, I made the change.